src/CFamily/CMake/MacroEvaluator.cs

```diff
@@ -58,33 +46,44 @@ internal MacroEvaluator(IEnvironmentVariableProvider environmentVariableProvider
             this.environmentVariableProvider = environmentVariableProvider;
         }
 
-        public string TryEvaluate(string macroPrefix, string macroName, EvaluationContext evaluationContext)
+        public string TryEvaluate(string macroPrefix, string macroName, IEvaluationContext evaluationContext)
         {
             if (string.IsNullOrEmpty(macroName))
             {
                 return null;
             }
 
-            // TODO:
-            // * process other simple macros
             if (macroPrefix != string.Empty)
             {
                 return macroPrefix.Equals("env", StringComparison.CurrentCultureIgnoreCase)
                     ? environmentVariableProvider.TryGet(macroName)
                     : null;
             }
 
-            if (macroName == "projectDir")
+            switch (macroName)
             {
-                return evaluationContext.RootDirectory;
+                case "workspaceRoot":
+                    return evaluationContext.RootDirectory;
+                case "projectFile":
+                    return evaluationContext.RootCMakeListsFilePath;
+                case "projectDir":
+                    return Path.GetDirectoryName(evaluationContext.RootCMakeListsFilePath);
+                case "projectDirName":
+                {
+                    var directoryName = Path.GetDirectoryName(evaluationContext.RootCMakeListsFilePath);
+                    return string.IsNullOrEmpty(directoryName) ? null : new DirectoryInfo(directoryName).Name;
+                }
+                case "thisFile":
```

Comment:
Technically all the file arguments don't make sense (`thisFile`, `projectFile`) in the buildRoot context. I don't know if it would be better to leave them and let it fail. VS would fail as well.

Reply:
Might as well leave them.